test/testsyncjournaldb.cpp

```diff
@@ -183,6 +183,58 @@ private slots:
         QCOMPARE(record.numericFileId(), QByteArray("123456789"));
     }
 
+    void testAvoidReadFromDbOnNextSync()
+    {
+        auto invalidEtag = QByteArray("_invalid_");
+        auto initialEtag = QByteArray("etag");
+        auto makeEntry = [&](const QByteArray &path, int type) {
+            SyncJournalFileRecord record;
+            record._path = path;
+            record._type = type;
+            record._etag = initialEtag;
+            _db.setFileRecord(record);
+        };
+        auto getEtag = [&](const QByteArray &path) {
+            SyncJournalFileRecord record;
+            _db.getFileRecord(path, &record);
+            return record._etag;
+        };
+
+        makeEntry("foodir", 2);
+        makeEntry("otherdir", 2);
+        makeEntry("foodir/file", 0);
+        makeEntry("foodir/subdir", 2);
+        makeEntry("foodir/subdir/file", 0);
+        makeEntry("foodir/otherdir", 2);
+        makeEntry("foodir/subdir/subsubdir", 2);
+        makeEntry("foodir/subdir/subsubdir/file", 0);
+        makeEntry("foodir/subdir/otherdir", 2);
```

Comment:
Was this test failling without the patch?

The test could folders file with _ and %